### PR TITLE
Add cols to cached_appeal_attributes for pagination

### PR DESCRIPTION
**Pull request by rubaiyat22**
connects #13141

### Description
- add two columns to cached_appeal_attributes table: `power_of_attorney_name` and `suggested_hearing_location` which will be used for filtering in hearings table
- add the newly created columns to update_cached_appeals_attributes_job
- create serializer for suggested_hearing_location which will be used in frontend in pagination work

### Database Changes
*Only for Schema Changes*

* [ ] Timestamps (created_at, updated_at) for new tables
* [x] Column comments updated
* [ ] Query profiling performed (eyeball Rails log, check bullet and fasterer output)
* [ ] Appropriate indexes added (especially for foreign keys, polymorphic columns, and unique constraints)
* [x] #appeals-schema notified with summary and link to this PR

### CODE REVIEW

File: .reek.yml [hunk 1]
```diff
@@ -74,6 +74,7 @@ detectors:
       - VeteranAttributeCacher
       - WarmBgsCachesJob
       - ExternalApi::PexipService#send_pexip_request
+      - UpdateCachedAppealsAttributesJob#format_suggested_hearing_location
```
Comment (rubaiyat22) [inline .reek.yml]: Ignoring feature_envy codeclimate error for now. Still thinking of how to resolve it

File: db/schema.rb [hunk 2]
```diff
@@ -10,7 +10,7 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 20200102193042) do
+ActiveRecord::Schema.define(version: 20200127223143) do
```
Comment (pkarman) [inline db/schema.rb]: might need merge conflict resolved -- be sure the result is the highest value.

File: app/models/hearings/available_hearing_locations.rb [hunk 3]
```diff
@@ -3,4 +3,25 @@
...
+    when "va_benefits_facility"
+      return "(BVA) " if loc.facility_id == "vba_372"
+
+      ro_or_vba
+    else
+      ""
+    end
+  end
+
+  def ro_or_vba
```
Comment (pkarman) [inline app/models/hearings/available_hearing_locations.rb]: maybe `determine_vba_facility_type` would be a more meaningful name?

File: .reek.yml (hunk 1 above)
Comment (pkarman) [reply .reek.yml]: Probably should be a method on the `appeal` maybe with a decorator pattern? see https://github.com/department-of-veterans-affairs/caseflow/wiki/Backend-Pattern-Decorator

File: app/models/hearings/available_hearing_locations.rb [hunk 4]
```diff
@@ -3,4 +3,30 @@
...
+  # this will eventually replace getFacilityType in AppealHearingLocations
+  # in upcoming pagination work
+  def formatted_facility_type
+    case facility_type
+    when "vet_center"
+      "(Vet Center) "
+    when "va_health_facility"
+      "(VHA) "
+    when "va_benefits_facility"
+      return "(BVA) " if facility_id == "vba_372"
```
Comment (ferristseng) [inline app/models/hearings/available_hearing_locations.rb]: Small nit pick, but these all have a space at the end of the string that I'm not sure is needed

File: db/migrate/20200127223143_add_poa_name_and_suggested_hearing_loc_to_cached_appeal_attributes.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+class AddPoaNameAndSuggestedHearingLocToCachedAppealAttributes < ActiveRecord::Migration[5.1]
+  def change
+    add_column :cached_appeal_attributes, :power_of_attorney_name, :string, comment: "'Firstname, Lastname' of power of attorney"
+    add_column :cached_appeal_attributes, :suggested_hearing_location, :string, comment: "Suggested hearing location in 'City, State (Facility Type)' format"
```
Comment (ferristseng) [inline db/migrate/20200127223143_add_poa_name_and_suggested_hearing_loc_to_cached_appeal_attributes.rb]: Nice comments for the new columns!

Small nit though is that I think it should be 'Firstname Lastname' for the POA (no comma)

File: app/serializers/helpers/appeal_hearing_helper.rb [hunk 6]
```diff
@@ -24,4 +25,9 @@ def hearings(appeal)
       AppealHearingSerializer.new(hearing).serializable_hash[:data][:attributes]
     end
   end
+
+  # returns the closest hearing location
+  def suggested_hearing_location(appeal)
+    available_hearing_locations(appeal).min_by { |loc| loc[:distance] }
```
Comment (ferristseng) [inline app/serializers/helpers/appeal_hearing_helper.rb]: One suggestion: `suggested_hearing_location` could live on the `Appeal` and `LegacyAppeal` as a shared concern instead so this logic isn't repeated here and in the update job.

**Pull request merged**